pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionActioner.java

```diff
@@ -153,9 +153,13 @@ RuntimeSpawner getRuntimeSpawner(Function.Instance instance, String packageFile)
         int instanceId = instance.getInstanceId();
 
         FunctionDetails.Builder functionDetailsBuilder = FunctionDetails.newBuilder(functionMetaData.getFunctionDetails());
+        
+        Function.FunctionAuthenticationSpec functionAuthenticationSpec
+                = instance.getFunctionMetaData().hasFunctionAuthSpec()
```

Comment:
Not sure if will fix the issue. Because as i see it, instance.getFunctionmetaData().hasFunctionAuthSpec() will always be true. The thing is that the condition check inside componentImpl who sets this is not right imo

Reply:
The logic in ComponentImpl is correct. clientAuthenticationDataHttps will be null if authentication is turned off because the authentication filter would not be added to the servlet